modules/articles/client/controllers/list-articles.client.controller.js

```diff
@@ -9,7 +9,45 @@
 
   function ArticlesListController(ArticlesService) {
     var vm = this;
+    var rightNow = new Date(Date.now());
 
-    vm.articles = ArticlesService.query();
+    vm.pageChanged = pageChanged;
+
+    init();
+
+    function init() {
+      vm.itemsPerPage = 20;
+      vm.currentPage = 1;
+
+      vm.statistics = ArticlesService.statistics();
+
+      find();
+    }
+
+    function pageChanged() {
+      find();
+    }
+
+    function find() {
+      search(vm.itemsPerPage, vm.currentPage)
+        .then(onSearchSuccess)
+        .catch(onSearchError);
+    }
+
+    function search(pageSize, page, createdAfter) {
+      return ArticlesService.query({
+        take: pageSize,
+        page: page,
+        //createdAfter: createdAfter || rightNow.setDate(rightNow.getDate() - 730)
+      }).$promise;
+    }
+
+    function onSearchSuccess(response) {
+      vm.articles = response;
+    }
+
+    function onSearchError(response) {
+      // handle error
```

Comment:
I was going to add `vm.error = response.data.message;` I was hoping to get feedback before I built this out too much on the front-end.